### PR TITLE
Allow get_export_names to skip configuration check

### PR DESCRIPTION
**Pull request by rmoe**
This introduces an environment variable named
NBCONVERT_DISABLE_CONFIG_EXPORTERS that will cause get_export_names to
return all entrypoints instead of checking the "enabled" status of
each one.

Closes: #1439

### CODE REVIEW

File: nbconvert/exporters/base.py
Comment (mriedem) [inline nbconvert/exporters/base.py:139]: nit: might be good to have a comment here so people know what this is for, i.e. bypass the (potentially slow) enable/disable check per extension if you know that you're supporting all exporters.

I also wonder if there is somewhere in the docs or config options that we could mention this environment variable otherwise people would only ever know about it if they were familiar with this code. Like maybe somewhere in here:

https://github.com/jupyter/nbconvert/blob/master/docs/source/api/exporters.rst

Though that's just auto-generated module doc.

Looking at https://nbconvert.readthedocs.io/en/latest/config_options.html#exporter-options maybe `Exporter.enabled` should mention this env var? Or https://nbconvert.readthedocs.io/en/latest/external_exporters.html?

A CODEOWNER might have better ideas, but I'd think mentioning it with the `Exporter.enabled` config option would be the most _discoverable_ place for mentioning this variable.

**Comment by rmoe** [comment]
@MSeal is there anything I need to do to help move this forward?

**Comment by MSeal** [comment]
@rmoe Sorry I originally missed this PR (was on leave). Let me leave a comment in the implementation, but I think a quick log message and adding a note to the documentation is all that's needed.

File: nbconvert/exporters/base.py
Comment (MSeal) [reply nbconvert/exporters/base.py:139]: To @mriedem 's point, https://nbconvert.readthedocs.io/en/latest/config_options.html#exporter-options would be the best place to document the environment variable action. You can add this note by editing the `CLI Flags and Aliases` section in `autogen_config.py`.

Additionally we should probably log out a quick message when we're disabling config exporters so it's visible that the flag was set when helping future nbconvert issues. Use `from traitlets.log import get_logger` and `get_logger().info("Config exporter loading disabled, no additional exporters will be automatically included.")`

**Comment by rmoe** [comment]
> @rmoe Sorry I originally missed this PR (was on leave). Let me leave a comment in the implementation, but I think a quick log message and adding a note to the documentation is all that's needed.

No worries, thank you for taking a look at this.

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/current-state-of-nbconvert-drop-support-for-python-3-6/8882/1

**Comment by mriedem** [comment]
@rmoe CI should be fixed on `master` now if you want to rebase.

File: nbconvert/exporters/tests/test_exporter.py
Comment (mriedem) [inline nbconvert/exporters/tests/test_exporter.py:100]:
```suggestion
        # In the absence of this variable we should revert to
```

File: nbconvert/exporters/tests/test_exporter.py
Comment (mriedem) [inline nbconvert/exporters/tests/test_exporter.py:89]: Fun fact, you can use `patch.dict` to temporarily set values in `os.environ`, there is an example in the docs:

https://docs.python.org/3.7/library/unittest.mock.html#unittest.mock.patch.dict

**Review by mriedem — APPROVED** [review]
Code looks good. A couple of small nits inline but the test looks good and there is a mention of this in the docs now (so comments have been resolved) and CI is passing. 👍

**Review by mriedem — APPROVED** [review]

**Review by MSeal — APPROVED** [review]
Thanks for getting this updated! LGTM

**Comment by MSeal** [comment]
doc build issue is unrelated, I'll look into that before the next release.

**Pull request merged**
